packages/core/helper-plugin/src/features/StrapiApp.tsx

```diff
@@ -109,9 +109,9 @@ const StrapiAppContext = React.createContext<StrapiAppContextValue>({
   settings: {},
   // These functions are required but should not resolve to undefined as they do here
   runHookParallel: () => Promise.resolve(),
-  runHookWaterfall: () => Promise.resolve(),
+  runHookWaterfall: (_name, data) => data,
```

Comment:
it doesn't need to return a promise anymore?

Reply:
Not by default, the actual function can return either, but it being default draws a lot of issues, but realistically it's default is never used anyway because the hook expects to be in the application.